pkg/virtual/workspaces/builder/build.go

```diff
@@ -86,35 +107,78 @@ func BuildVirtualWorkspace(rootPathPrefix string, clusterWorkspaces workspaceinf
 				AddToScheme:        tenancyv1beta1.AddToScheme,
 				OpenAPIDefinitions: kcpopenapi.GetOpenAPIDefinitions,
 				BootstrapRestResources: func(mainConfig genericapiserver.CompletedConfig) (map[string]fixedgvs.RestStorageBuilder, error) {
-					reviewerProvider := workspaceauth.NewAuthorizerReviewerProvider(subjectLocator)
-					workspaceAuthorizationCache = workspaceauth.NewAuthorizationCache(
-						clusterWorkspaces.Lister(),
-						clusterWorkspaces.Informer(),
-						reviewerProvider.ForVerb("get"),
-						rbacInformers,
-					)
 
-					workspaceClient := orgKcpClient.TenancyV1alpha1().ClusterWorkspaces()
-					clusterWorkspaceCache := workspacecache.NewClusterWorkspaceCache(
-						clusterWorkspaces.Informer(),
-						workspaceClient,
+					rootTenancyClient := rootKcpClient.TenancyV1alpha1()
+					rootRBACClient := rootKubeClient.RbacV1()
+
+					globalClusterWorkspaceCache = workspacecache.NewClusterWorkspaceCache(
+						wildcardsClusterWorkspaces.Informer(),
+						func(orgName string) tenancyv1alpha1.TenancyV1alpha1Interface {
+							return tenancyv1alpha1.NewWithCluster(rootTenancyClient.RESTClient(), orgName)
+						},
 						"")
 
+					rootRBACInformers := rbacwrapper.FilterInformers(helper.RootCluster, wildcardsRbacInformers)
+					rootSubjectLocator := frameworkrbac.NewSubjectLocator(rootRBACInformers)
+					rootReviewerProvider := workspaceauth.NewAuthorizerReviewerProvider(rootSubjectLocator)
+
+					rootClusterWorkspaceInformer := tenancywrapper.FilterClusterWorkspaceInformer(helper.RootCluster, wildcardsClusterWorkspaces)
+
+					rootWorkspaceAuthorizationCache = workspaceauth.NewAuthorizationCache(
+						rootClusterWorkspaceInformer.Lister(),
+						rootClusterWorkspaceInformer.Informer(),
+						rootReviewerProvider.ForVerb("get"),
+						rootRBACInformers,
+					)
+
+					adminUser := &user.DefaultInfo{
```

Comment:
this need a comment. Looks like a security time bomb.

Reply:
Fixed in commit e961ccc7314f96a5b76132dd8455350bfca62069